pkg/kapp/cmd/app/deploy.go

```diff
@@ -225,8 +227,25 @@ func (o *DeployOptions) Run() error {
 		}
 
 		// Remove unused GVs and GKs
-		return app.UpdateUsedGVsAndGKs(failingAPIServicesPolicy.GVs(newResources, nil),
+		err = app.UpdateUsedGVsAndGKs(failingAPIServicesPolicy.GVs(newResources, nil),
 			NewUsedGKsScope(newResources).GKs())
+		if err != nil {
+			return err
+		}
+
+		if o.DeployFlags.AppMetadataFile != "" {
+			meta, err := app.Meta()
+			if err != nil {
+				return err
+			}
+			err = ioutil.WriteFile(o.DeployFlags.AppMetadataFile, []byte(meta.AsString()), os.ModePerm)
+			if err != nil {
+				return err
+			}
```

Comment:
sgtm! returned an err instead of nil at the end

Reply:
hehe, I actually meant that we can remove the err check and just return err at the end instead of nil. 
```go
			err = os.WriteFile(o.DeployFlags.AppMetadataFile, []byte(meta.AsString()), os.ModePerm)
		}
		return err
```
But I am okay with the current way as well because err should be handled closer to it’s  origin.